oauth2_provider/models.py

```diff
@@ -501,16 +502,27 @@ def revoke(self):
         """
         access_token_model = get_access_token_model()
         refresh_token_model = get_refresh_token_model()
-        with transaction.atomic():
+
+        access_token_database = router.db_for_write(access_token_model)
+        refresh_token_database = router.db_for_write(refresh_token_model)
+
+        # This is highly unlikely, but let's warn people just in case it does.
```

Comment:
Does it make sense to make it a requirement that access and refresh token databases must be the same?

Reply:
Yes, it likely does make sense. This could be extracted into a runtime check with a clear failure along with a documentation update.